src/dataset.jl

```diff
@@ -1,3 +1,29 @@
+
+"""
+    KeyAlignmentError
+
+Is thrown when the constrained dimensions of components in a `KeyedDataset` have misaligned
+key values.
+
+# Fields
+* constraint::Pattern - The constraint descript all patch dimension keys
+* groups - An iterator of paths and keys for each non-matching group.
+"""
+struct KeyAlignmentError <: Exception
+    constraint::Pattern
+    groups
```

Comment:
```suggestion
    groups::Tuple{Vector{Tuple}}
```
is it worth assigning this?

Reply:
Well this would just be a `Zip` iterator and it probably wouldn't be worth calling `collect` first.